src/main/java/com/faforever/client/fx/contextmenu/ChatUserColorPickerCustomMenuItemController.java

```diff
@@ -0,0 +1,84 @@
+package com.faforever.client.fx.contextmenu;
+
+import com.faforever.client.chat.ChatChannelUser;
+import com.faforever.client.chat.ChatColorMode;
+import com.faforever.client.chat.ChatUserCategory;
+import com.faforever.client.chat.event.ChatUserColorChangeEvent;
+import com.faforever.client.fx.JavaFxUtil;
+import com.faforever.client.preferences.ChatPrefs;
+import com.faforever.client.preferences.PreferencesService;
+import com.faforever.client.util.Assert;
+import com.google.common.eventbus.EventBus;
+import javafx.beans.WeakInvalidationListener;
+import javafx.scene.control.Button;
+import javafx.scene.control.ColorPicker;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.Locale;
+
+import static com.faforever.client.chat.ChatColorMode.RANDOM;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+@Slf4j
+public class ChatUserColorPickerCustomMenuItemController extends AbstractCustomMenuItemController<ChatChannelUser> {
+
+  public ColorPicker colorPicker;
+  public Button removeCustomColorButton;
+
+  private final PreferencesService preferencesService;
+  private final EventBus eventBus;
+
+  private ChatPrefs chatPrefs;
+
+  @Override
+  public void initialize() {
+    chatPrefs = preferencesService.getPreferences().getChat();
+    removeCustomColorButton.setOnAction(event -> colorPicker.setValue(null));
+    JavaFxUtil.bindManagedToVisible(removeCustomColorButton);
+    WeakInvalidationListener weakChatColorModePropertyListener = new WeakInvalidationListener((observable) -> JavaFxUtil.runLater(() -> {
+      ChatColorMode chatColorMode = chatPrefs.getChatColorMode();
+      removeCustomColorButton.setVisible(!chatColorMode.equals(RANDOM) && colorPicker.getValue() != null);
+      getRoot().setVisible(!chatColorMode.equals(RANDOM));
+    }));
```

Comment:
The underlying listener here needs to be a field of the class otherwise it will be available for garbage collection as soon as the method exits and so could get removed before being triggered